### PR TITLE
Dynamic type mapping for Put collection of objects

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Part of #371.

@nikitin-da PTAL

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
BTW, I decided to group tests by inner classes

File: storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/operation/put/PreparedPutCollectionOfObjects.java [hunk 1]
```diff
@@ -76,19 +124,22 @@
             if (useTransaction) {
                 storIOSQLite.internal().endTransaction();
 
+                // if delete was in transaction and it was successful -> notify about changes
```
Comment (nikitin-da) [inline storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/operation/put/PreparedPutCollectionOfObjects.java]: put =)

File: storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/operation/put/PreparedPutCollectionOfObjectsTest.java [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(Enclosed.class)
+public class PreparedPutCollectionOfObjectsTest {
+
+    public static class WithoutTypeMapping {
```
Comment (nikitin-da) [inline storio-sqlite/src/test/java/com/pushtorefresh/storio/sqlite/operation/put/PreparedPutCollectionOfObjectsTest.java]: Nice decision!

File: storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/operation/put/PreparedPutCollectionOfObjects.java (hunk 1 above)
Comment (artem-zinnatullin) [inline storio-sqlite/src/main/java/com/pushtorefresh/storio/sqlite/operation/put/PreparedPutCollectionOfObjects.java]: Damn, copy-paster!

**Pull request merged**